.github/workflows/generate-translation-files.yml

```diff
@@ -0,0 +1,76 @@
+# This workflow generates translation files from code and moves the generated
+# translation files to the openedx-translations repository via an auto-merged pull
+# request.
+
+name: Generate Translation Files
+
+on:
+  # This action is meant to be used from other actions.
+  # https://docs.github.com/en/actions/learn-github-actions/reusing-workflows
+  - workflow_call
+
+jobs:
+  python-translations:
+    runs-on: ubuntu-20.04
+
+    env:
+      REPO_NAME: ${{ github.event.repository.name }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # Installs gettext, a dependency for generating translation files in django
+      - name: install gettext
+        run: sudo apt install -y gettext
+
+      # Sets up Python
+      - name: setup python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.8'
+
+      # Installs Python requirements from translations.txt
+      - name: install requirements
+        run: pip install -r requirements/translations.txt
+
+      # Generates the english translation files
+      - name: generate translations
+        run: make extract_translations
+
+      # Clones the openedx-translations repository
+      - name: clone openedx-translations
+        uses: actions/checkout@v3
+        with:
+          repository: openedx/openedx-translations
+          path: openedx-translations
+
+      # Copies the generated translation files to the openedx-translations repo
+      - name: copy translation files to openedx-translations
+        run: |
+          EN_DIR=$(find . -type d -name "en")
+          mkdir openedx-translations/$REPO_NAME
```

Comment:
my assumption reading this line: `mkdir` must be run the first time this time runs in a repo; it is not an issue to continue to run it once the dir has been established